File: OneDrive/Desktop/Intro_DSA/groupassignment/requirements.cpp
#include <iostream>
#include <vector>  // Required for vector container
#include <limits>  // Required for numeric_limits

using namespace std;

int get_number_of_elements() {
    int num_elements;
    while (true) {
        try {
            cout << "Enter the number of elements in the array: ";
            cin >> num_elements;
            if (num_elements < 0) {
                throw invalid_argument("The number of elements must be a positive integer.");
            }
            return num_elements;
        } catch (const invalid_argument& e) {
            cout << e.what() << endl;
            // Clear input buffer in case of error
            cin.clear();
            cin.ignore(numeric_limits<streamsize>::max(), '\n');
        }
    }
}

vector<int> get_elements(int num_elements) {
    vector<int> arr;
    for (int i = 0; i < num_elements; ++i) {
        while (true) {
            try {
                int element;
                cout << "Enter element " << i + 1 << ": ";
                cin >> element;
                if (cin.fail()) {
                    cin.clear(); // Clear input error state
                    cin.ignore(numeric_limits<streamsize>::max(), '\n'); // Ignore invalid input
                    throw invalid_argument("Please enter a valid integer.");
                }
                arr.push_back(element);
                break; // Exit the inner loop once a valid element is entered
            } catch (const invalid_argument& e) {
                cout << e.what() << endl;
                // Clear input buffer in case of error
                cin.clear();
                cin.ignore(numeric_limits<streamsize>::max(), '\n');
            }
        }
    }
    return arr; // Return the vector of elements
}

File: OneDrive/Desktop/Intro_DSA/groupassignment/requirements.py
def get_number_of_elements():
    """
    Prompts the user to input the number of elements for the array.

    Returns:
    int: The number of elements in the array.
    """
    while True:
        try:
            num_elements = int(input("Enter the number of elements in the array: "))
            if num_elements < 0:
                raise ValueError("The number of elements must be a positive integer.")
            return num_elements
        except ValueError as e:
            print(e)

def get_elements(num_elements):
    """
    Prompts the user to input the elements of the array.

    Parameters:
    num_elements (int): The number of elements to input.

    Returns:
    list: The list of input elements.
    """
    arr = []
    for i in range(num_elements):
        while True:
            try:
                element = int(input(f"Enter element {i + 1}: "))
                arr.append(element)
                break
            except ValueError:
                print("Please enter a valid integer.")
    return arr
